### PR TITLE
Move deploy & run to the status bar and reduce number of toolbar buttons

### LINKED ISSUE #2318: Improve status bar
__What should we do?__

Implement missing pieces of the status bar story.

Activities:

* [x] move deploy button to the status bar
  * [x] status bar button opens the modal
  * [x] status bar button is placed correctly
* [x] move run button to the status bar
  * [x] status bar button opens the modal
  * [x] status bar button is placed correctly
* [x] reduce buttons number on the toolbar

Design: https://github.com/camunda/camunda-modeler-design/issues/1

Already done:

* [x] engine profile (already done)
* [x] xml button
* [x] log button

__Why should we do it?__

It's part of [the status bar vision](https://github.com/camunda/camunda-modeler-design/blob/main/visual-ide/STATUS_BAR.md) and a step towards the Visual IDE goal.

### PR DESCRIPTION
**Pull request by barmac**
This implements basic concept for deploy & run on the status bar and reduces number of buttons on the toolbar.

Closes #2318

### CODE REVIEW

**Comment by philippfromme** [comment]
Is it intentional that almost all buttons are gone when the _Welcome_ tab is open?

image

**Comment by philippfromme** [comment]
So the idea is that the open button is gone from the toolbar as it is accessible through the application menu?

**Comment by barmac** [comment]
> So the idea is that the open button is gone from the toolbar as it is accessible through the application menu?

Yup. Removal of the buttons which are available via the application menu is [part of the design](https://github.com/camunda/camunda-modeler-design/issues/1#issuecomment-866713412).

**Comment by philippfromme** [comment]
There are no hover styles except for the cursor when hovering the dropdown.

SUqHJeRjjB

For comparison:

mKaD7nVanU

Is this intentional?

**Comment by andreasgeier** [comment]
Good point. Please use a very light grey from our color variables to show a hover state.

**Comment by barmac** [comment]
I've re-styled the dropdown so it has a light grey background on hover. I also reduced the padding a bit since it didn't look right with the background color.

https://user-images.githubusercontent.com/28307541/123598719-17baa780-d7f5-11eb-8e9b-b70569c6bc38.mp4

**Comment by barmac** [comment]
This PR is available as build artifacts via `2318-status-bar` build.

File: client/src/shared/ui/overlay/OverlayDropdown.js [hunk 1]
```diff
@@ -76,8 +74,8 @@ function Option(props) {
   const { text, onClick } = props;
 
   return (
-    <li>
-      <button onClick={ onClick } type="button">{ text }</button>
+    <li onClick={ onClick }>
```
Comment (philippfromme) [inline client/src/shared/ui/overlay/OverlayDropdown.js:77]: Is this part of the re-styling?

Comment (nikku) [reply client/src/shared/ui/overlay/OverlayDropdown.js:77]: Proposal: Always use semantic elements as click targets (`button`, `link`).

**Review by andreasgeier — APPROVED** [review]
✅ &nbsp; LGTM

File: client/src/shared/ui/overlay/OverlayDropdown.js (hunk 1 above)
Comment (barmac) [reply client/src/shared/ui/overlay/OverlayDropdown.js:77]: So the idea here is that you can click anywhere on the list item and it triggers a click. Other solution for that would be to stretch the button to fill the whole space of `<li>`. Not sure if this is better. BTW this is similar to [what we already have](https://github.com/camunda/camunda-modeler/blob/develop/client/src/shared/ui/DropdownButton.js#L139).

**Review by philippfromme — APPROVED** [review]
👏🏻

**Pull request merged**